scikeras/_utils.py

```diff
@@ -231,8 +240,113 @@ def has_param(func: Callable, param: str) -> bool:
 
 
 def accepts_kwargs(func: Callable) -> bool:
+    """Check if ``func`` accepts kwargs.
+    """
     return any(
         True
         for param in inspect.signature(func).parameters.values()
         if param.kind == param.VAR_KEYWORD
     )
+
+
+def compile_with_params(items, params, base_params=None):
```

Comment:
Could you elaborate a bit? The signature of `params_w_objs = render(self.get_params())` does not match the current signature of `compile_with_params`.

I think you are proposing `render` since that alludes to how HTML (and similar nested structures) are rendered? Or is there some other reason I'm missing? It doesn't stand out to me as an obvious choice.

Reply:
If you have a strong feeling about either of those, please assert which one and I'll be happy to change it. Personally I feel that `compile_with_params` is more explicit but I am often wrong.

Reply:
`parse` is the right name.

Reply:
`parse` sounds good, but I think it should be a bit more descriptive than a single word, how bout `parse_nested_params` or something like that?

Reply:
> parse_nested_params

Now I like "unflatten". The input is a flat dict, the output is a "nested" dict (each element has own set of values).

Reply:
`unflatten_params`? `expand_params`? `expand_routed_params`?

Reply:
I like "unflatten_params" best. It describes the input (flat params) and action (unflatten).

Reply:
Sounds good to me. I'll rename this tomorrow, and then let's get this merged?

Reply:
Yeah, I think it's time to merge. Then #79.